.github/workflows/ci_pull_request_open.yml

```diff
@@ -0,0 +1,63 @@
+name: Pull Request Open
+on:
+  pull_request:
+jobs:
+  ci-build-test-pr:
+    runs-on: ubuntu-latest
+    env:
+      MAVEN_OPTS: "-Xmx1500m -XX:+HeapDumpOnOutOfMemoryError"
+      GH_TOKEN: ${{ github.token }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Java
+        uses: actions/setup-java@v4
+        with:
+          java-version: 21
+          distribution: temurin
+          cache: maven
+
+      - uses: graalvm/setup-graalvm@v1
+        with:
+          java-version: '21'
+          distribution: 'graalvm'
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - run: ./mvnw --version
+
+      - run: >
+            ./mvnw -s maven-settings.xml clean install -DskipTests -Pnative
+
+      - name: Test with Maven
+        run: >
+            ./mvnw -s maven-settings.xml verify -B -e -DrerunFailingTestsCount=2 -Pnative
+            -fae -Dansi.strip=true
+
+      - name: Publish Test Report
+        if: success() || failure()
+        uses: ScaCap/action-surefire-report@v1
+        with:
+          fail_on_test_failures: true
```

Comment:
I think we want this to be `false` as the test step will already fail if we have failures, but we want the reports to still be published.

Reply:
I was torn on this: sure we need failure report, but is it ok to act on flaky jiras in presence of test failure?

Reply:
A flaky failure is still a flaky test, regardless of whether other tests failed, so I think it's better to report these events.